src/main/kotlin/frc/robot/subsystems/drive/ProfiledPosePID.kt

```diff
@@ -114,14 +131,27 @@ fun setTolerance(pose2d: Pose2d) {
     thetaController.setTolerance(pose2d.rotation.radians)
 }
 
+private fun updateAccelerationConstraints() {
+    xController.constraints = linearLimits
+    yController.constraints = linearLimits
+    thetaController.constraints = rotationalLimits
+}
+
 /**
  * Returns field relative chassis speeds to the selected goal.
  * @botPose the current pose of the robot
  */
-fun getSpeedSetpoint(botPose: Pose2d): () -> ChassisSpeeds = {
-    ChassisSpeeds(
-        xController.calculate(botPose.x),
-        yController.calculate(botPose.y),
-        thetaController.calculate(botPose.rotation.radians)
-    )
+fun getSpeedSetpoint(botPose: Pose2d): () -> ChassisSpeeds {
+    cycleCounter++
+    if (cycleCounter >= ACCELERATION_UPDATE_CYCLES) {
+        updateAccelerationConstraints()
+        cycleCounter = 0
+    }
```

Comment:
```suggestion
    Logger.runEveryN(ACCELERATION_UPDATE_CYCLES, ::updateAccelerationConstraints)
```
https://docs.advantagekit.org/javadoc/org/littletonrobotics/junction/Logger.html#runEveryN(int,java.lang.Runnable)

Reply:
Also, won't the robot stutter if doing this after the angle stabilizes?

Reply:
Seemed fine when i tested it yesterday